src/concurrency/timestamp_ordering_transaction_manager.cpp

```diff
@@ -93,6 +93,10 @@ Transaction *TimestampOrderingTransactionManager::BeginTransaction() {
   auto eid = EpochManagerFactory::GetInstance().EnterEpoch(begin_cid);
   txn->SetEpochId(eid);
 
+  auto &log_manager = logging::LogManager::GetInstance();
+  log_manager.PrepareLogging();
+  log_manager.LogBeginTransaction(begin_cid);
```

Comment:
If a transaction abort, the only log record it will generate it's a begin transaction record. It won't log down any tuple record because we only do that when committing a transaction.